src/main/java/duke/Command.java

```diff
@@ -0,0 +1,252 @@
+                ui.printError("Unable to write to file.");
+            }
+            break;
+        }
+    }
+
+    private boolean isValidDate(String date) {
+        try {
+            LocalDate d = LocalDate.parse(date, DateTimeFormatter.ofPattern(INPUT_DATE_FORMAT));
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
+
+    private boolean isCorrectUsage(String args, Duke duke) {
+        UI ui = duke.getUI();
+        TaskList taskList = duke.getTaskList();
+        String usage = getCorrectUsage();
+        switch(name) {
+        case "delete":
+            if (args.isEmpty()) {
+                ui.printError("Please specify a task number.\n\n" + usage);
+                return false;
+            }
+            return true;
+
+        case "find":
+            if (args.isBlank()) {
+                ui.printError("Please specify at least one keyword.\n\n" + usage);
+                return false;
+            } else if (args.charAt(0) != '"' || args.charAt(args.length() - 1) != '"') {
+                if (args.split(" ").length > 1) {
+                    ui.printError("'find' only expects 1 argument.\n\n If there are multiple "
+                            + "keywords, please enclose them in quotation marks (\"\").");
+                    return false;
+                }
+            } else {
+                if (args.length() == 2 || args.substring(1, args.length() - 1).isBlank()) {
+                    ui.printError("Please specify at least one keyword.\n\n" + usage);
+                    return false;
+                }
+            }
+            return true;
+
+        case "list":
+            if (!args.isEmpty()) {
+                ui.printError("'list' expects no arguments.");
+                return false;
+            }
+            return true;
+
+        case "deadline":
+            String[] temp = args.split(" /by ", 2);
+            if (temp.length < 2) {
+                ui.printError("Please specify a task and deadline.\n\n" + usage);
+                return false;
+            } else if (!isValidDate(temp[1])) {
+                ui.printError("Please specify the due date in the right format.\n\n" + usage);
+                return false;
+            }
+            return true;
+
+        case "event":
+            temp = args.split(" /at ", 2);
+            if (temp.length < 2) {
+                ui.printError("Please specify an event and date.\n\n" + usage);
+                return false;
+            } else if (!isValidDate(temp[1])) {
+                ui.printError("Please specify the event date in the right format.\n\n" + usage);
+                return false;
+            }
+            return true;
+
+        case "todo":
+            if (args.isEmpty()) {
+                ui.printError("Please specify a task.\n\n" + usage);
+                return false;
+            }
+            return true;
+
+        case "mark":
```

Comment:
You might need to add an explicit "// fallthrough" comment when you are using the same code for multiple cases